guardian/recipes-backend
# Support timed "launch" of curation data

## What does this change?

- Updates the recipe backend API to support dated curation delivery.
- Adds a lambda function to "launch" todays curation by copying it over the default path
  - Also detects when an update is made to todays curation and "re-launches" it

### New endpoints/arguments:

**POST** https://recipes.code.dev-guardianapis.com/api/curation/{region}/{variant}?date={YYYY-MM-DD} [authenticated]
**GET** https://recipes.code.dev-guardianapis/{region}/{variant}/{YYYY-MM-DD}/curation.json [open]

## How to test

1. POST some JSON to a 'test' region/variant combination, e.g. `region-andy-test`/`variant-one` using TOMORROW'S date.  Use a name that nobody else will use as you'll need to go back tomorrow.
2. Test that you can access it with the GET request above
3. Test that **GET** https://recipes.code.dev-guardianapis/{region}/{variant}/curation.json returns you 403
4. **POST** some _different_ JSON to the same region/variant combination, using TODAY'S date
5. **GET** https://recipes.code.dev-guardianapis/{region}/{variant}/curation.json should now return you this JSON (you might need to try a couple of times to allow for CDN lag)
6. Test the same GET tomorrow.  It should now return you the JSON you originally uploaded with tomorrow's date.

## How can we measure success?

Able to support future-launching of fronts

## Have we considered potential risks?

There is currently no protection against one user over-writing another's changes.  This will come with the Fronts Tool work.
